client/src/pages/Dashboard.jsx

```diff
@@ -1,31 +1,58 @@
 import React from "react";
+import axios from "axios";
+import {Redirect, useHistory } from "react-router-dom";
 import Button from "@material-ui/core/Button";
 import CssBaseline from "@material-ui/core/CssBaseline";
 import TextField from "@material-ui/core/TextField";
 import Paper from "@material-ui/core/Paper";
-import { useHistory } from "react-router-dom";
 
 export default function Dashboard() {
   const history = useHistory();
+  const [user, setUser] = React.useState(null)
+  const [loading, setLoading] = React.useState(true)
+
+  const getUser = async () => {
+    try {
+      const {data} = await axios.get('/users/auth');
+      localStorage.setItem("userInfo",JSON.stringify(data.user));
+    } catch (error) {
+      console.log(error.message);
+    }
+    setLoading(false);
+  };
+  
 
   React.useEffect(() => {
-    const user = localStorage.getItem("user");
-    if (!user) history.push("/signup");
-  }, []);
+    if(loading){
+      getUser();
+    }
+    const userInfo = JSON.parse(localStorage.getItem('userInfo'));
+    if (!userInfo && !loading){
+      history.push('/login');
```

Comment:
You might want to implement this at a higher level, such as by implementing a protected route. As is, there might be odd behavior like flashing the dashboard page before redirecting to the login page